From a803fa9daa63895fdc89501b99ac0bbcafcf45f5 Mon Sep 17 00:00:00 2001
From: Shivani Baranwal <quic_shivbara@quicinc.com>
Date: Tue, 3 Sep 2024 14:08:41 +0530
Subject: [PATCH] Include the NAN header file into driver.h to avoid C++
 constraints

Include src/common/nan.h file into src/drivers/driver.h to resolve the
compilation issue "ISO C++ forbids forward references to 'enum' types"
by pulling in the full definition of enum nan_service_protocol_type.

Signed-off-by: Shivani Baranwal <quic_shivbara@quicinc.com>
---
 src/drivers/driver.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/drivers/driver.h b/src/drivers/driver.h
index 30cbbc568..7ec5fb1e1 100644
--- a/src/drivers/driver.h
+++ b/src/drivers/driver.h
@@ -22,6 +22,7 @@
 #include "common/defs.h"
 #include "common/ieee802_11_defs.h"
 #include "common/wpa_common.h"
+#include "common/nan.h"
 #ifdef CONFIG_MACSEC
 #include "pae/ieee802_1x_kay.h"
 #endif /* CONFIG_MACSEC */
@@ -29,7 +30,6 @@
 
 struct nan_subscribe_params;
 struct nan_publish_params;
-enum nan_service_protocol_type;
 
 #define HOSTAPD_CHAN_DISABLED 0x00000001
 #define HOSTAPD_CHAN_NO_IR 0x00000002
-- 
2.43.0

